### DISCUSSION TITLE
Samba - Read only share

### DISCUSSION THREAD

**Discussion by rmervine** [retronas/discussions#16] [Q&A]
Built in to retronas we can set a username and password for the targeted samba share. I would like to consider adding a new user and give that user read only access to the retronas shares. That way I reduce the risk of accidental deletion. And  then I would manage the retronas username/password for adding items on demand.

Is it possible to do this? I was browsing the smb.conf and I see it references /etc/samba/retronas.conf. And I do see the warning that the conf file is autogenerated.

I'm thinking I could potentially do a read list, but then I'm thinking it's going to get overwritten like it says in the conf file.

**Comment by sairuk** [comment]
this can be done but we use samba configs for multiple services so i don't know if we would add it to the main project but you are correct the samba configs would be overridden on a reinstall. 

if you want to do your own you would do something like this as root, to your own file

```
# adduser --no-create-home --disabled-password --disabled-login  <username>
# smbpasswd -a <username>
# cat << EOF > /etc/samba/my-readonly-share.conf
[retronas-ro]
    comment = retronas read only share
    path = /data/retronas
    browsable = yes
    read only = yes
    valid users = <username>
    guest ok = no
EOF
```
add in the following lines at the bottom of the smb.conf to start using the share
```
# vim /etc/samba/smb.conf
[retronas-ro]
include = /etc/samba/my-readonly-share.conf
```

restart samba
```
systemctl restart smbd.service
```
